compiler/semexprs.nim

```diff
@@ -164,7 +164,7 @@ proc isCastable(conf: ConfigRef; dst, src: PType): bool =
   #                       tySequence, tyPointer, tyNil, tyOpenArray,
   #                       tyProc, tySet, tyEnum, tyBool, tyChar}
   let src = src.skipTypes(tyUserTypeClasses)
-  if skipTypes(dst, abstractInst-{tyOpenArray}).kind == tyOpenArray:
+  if skipTypes(dst, abstractInst-{tyOpenArray}).kind in {tyOpenArray}:
```

Comment:
Nitpick: Since you decided to go for another solution (compared to the initial one which had `{tyOpenArray, tyUncheckedArray}`), this change can be reverted to keep things clean.

Reply:
sure